cli/deploy-moe-vnet.sh

```diff
@@ -0,0 +1,81 @@
+export VNET_NAME=vnet-$SUFFIX
+export SUBNET_NAME="snet-scoring"
+export ENDPOINT_NAME=endpt-vnet-`echo $RANDOM`
+
+# Get the current branch name of the azureml-examples. Useful in PR scenario. Since the sample code is cloned and executed from a VM, we need to pass the branch name when running az vm run-command
+# If running from local machine, change it to your branch name
+export GIT_BRANCH=$GITHUB_HEAD_REF
+
+# We use a different workspace for managed vnet endpoints
+az configure --defaults workspace=$WORKSPACE
+
+### setup VM & deploy/test ###
+# if vm exists, wait for 15 mins before trying to delete
+export VM_EXISTS=$(az vm list -o tsv --query "[?name=='$VM_NAME'].name")
+if [ "$VM_EXISTS" != "" ];
+then
+    echo "VM already exists from previous run. Waiting for 15 mins before deleting."
+	sleep 15m
+	az vm delete -n $VM_NAME -y
+fi
+
+# create the VM
+az deployment group create --template-file endpoints/online/managed/vnet/test_scoring/vm-main.bicep --parameters vmName=$VM_NAME identityName=$IDENTITY_NAME vnetName=$VNET_NAME subnetName=$SUBNET_NAME
+
+## Note for doc: create a VM: az vm create -n $VM_NAME
+
+# command in script: az deployment group create --template-file endpoints/online/managed/vnet/setup/vm_main.bicep #identity name is hardcoded uai-identity 
+az vm run-command invoke -n $VM_NAME --command-id RunShellScript --scripts @endpoints/online/managed/vnet/test_scoring/scripts/vmsetup.sh --parameters "SUBSCRIPTION:$SUBSCRIPTION" "RESOURCE_GROUP:$RESOURCE_GROUP" "LOCATION:$LOCATION" "IDENTITY_NAME:$IDENTITY_NAME" "GIT_BRANCH:$GIT_BRANCH"
+
+# build image
+az vm run-command invoke -n $VM_NAME --command-id RunShellScript --scripts @endpoints/online/managed/vnet/test_scoring/scripts/build_image.sh --parameters "SUBSCRIPTION:$SUBSCRIPTION" "RESOURCE_GROUP:$RESOURCE_GROUP" "LOCATION:$LOCATION" "IDENTITY_NAME:$IDENTITY_NAME" "ACR_NAME=$ACR_NAME"
+
+# create endpoint/deployment inside managed vnet
+az vm run-command invoke -n $VM_NAME --command-id RunShellScript --scripts @endpoints/online/managed/vnet/test_scoring/scripts/create_moe.sh --parameters "SUBSCRIPTION:$SUBSCRIPTION" "RESOURCE_GROUP:$RESOURCE_GROUP" "LOCATION:$LOCATION" "IDENTITY_NAME:$IDENTITY_NAME" "WORKSPACE:$WORKSPACE" "ENDPOINT_NAME:$ENDPOINT_NAME" "ACR_NAME=$ACR_NAME"
+
+# test the endpoint by scoring it
+export CMD_OUTPUT=$(az vm run-command invoke -n $VM_NAME --command-id RunShellScript --scripts @endpoints/online/managed/vnet/test_scoring/scripts/score_endpoint.sh --parameters "SUBSCRIPTION:$SUBSCRIPTION" "RESOURCE_GROUP:$RESOURCE_GROUP" "LOCATION:$LOCATION" "IDENTITY_NAME:$IDENTITY_NAME" "WORKSPACE:$WORKSPACE" "ENDPOINT_NAME:$ENDPOINT_NAME")
+
+# the scoring output for sample request should be [11055.977245525679, 4503.079536107787]. We are validating if part of the number is available in the output (not comparing all the decimals to accomodate rounding discrepencies)
+if [[ $CMD_OUTPUT =~ "11055" ]]; then
```

Comment:
Do you want to rely on this or deserialize the response that has the provisioning status "Succeeded"

Reply:
Unfortunately "provisioning succedded" does not seem to indicate script failures. See here: https://github.com/Azure/azureml-examples/runs/5501656227?check_suite_focus=true#step:6:184